commit e539116767ba7bfab01e2b0f510b5b6b8f12d051
Author: Zbigniew JÄ™drzejewski-Szmek <zbyszek@in.waw.pl>
Date:   Fri Mar 11 17:06:17 2016 -0500

    yummy:patchX:resolved--create--etc-resolv-conf-symlink-at-runtime.patch

resolved: create /etc/resolv.conf symlink at runtime

    
    If the symlink doesn't exists, and we are being started, let's
    create it to provie name resolution.
    
    If it exists, do nothing. In particular, if it is a broken symlink,
    we cannot really know if the administator configured it to point to
    a location used by some service that hasn't started yet, so we
    don't touch it in that case either.
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1313085

diff --git a/src/resolve/resolved.c b/src/resolve/resolved.c
index deb75f9..914d3b8 100644
--- a/src/resolve/resolved.c
+++ b/src/resolve/resolved.c
@@ -67,6 +67,10 @@ int main(int argc, char *argv[]) {
                 goto finish;
         }
 
+        r = symlink("../run/systemd/resolve/resolv.conf", "/etc/resolv.conf");
+        if (r < 0 && errno != EEXIST)
+                log_warning_errno(errno, "Could not create /etc/resolv.conf symlink: %m");
+
         /* Drop privileges, but keep three caps. Note that we drop those too, later on (see below) */
         r = drop_privileges(uid, gid,
                             (UINT64_C(1) << CAP_NET_RAW)|          /* needed for SO_BINDTODEVICE */
diff --git a/tmpfiles.d/etc.conf.m4 b/tmpfiles.d/etc.conf.m4
index 064eae9..928105e 100644
--- a/tmpfiles.d/etc.conf.m4
+++ b/tmpfiles.d/etc.conf.m4
@@ -13,9 +13,6 @@ L+ /etc/mtab - - - - ../proc/self/mounts
 m4_ifdef(`HAVE_SMACK_RUN_LABEL',
 t /etc/mtab - - - - security.SMACK64=_
 )m4_dnl
-m4_ifdef(`ENABLE_RESOLVED',
-L! /etc/resolv.conf - - - - ../usr/lib/systemd/resolv.conf
-)m4_dnl
 C /etc/nsswitch.conf - - - -
 m4_ifdef(`HAVE_PAM',
 C /etc/pam.d - - - -
